tests/virt_autotest/validate_system_health.pm

```diff
@@ -0,0 +1,84 @@
+# SUSE's openQA tests
+#
+# Copyright 2012-2022 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+#
+# Summary: Do a basic examination to the host and guests after tests run
+# Maintainer: Julie CAO <JCao@suse.com>, qe-virt@suse.de
+
+use base "virt_feature_test_base";
+use strict;
+use warnings;
+use testapi;
+use utils;
+use ipmi_backend_utils qw(reconnect_when_ssh_console_broken);
+use Utils::Architectures;
+use virt_autotest::common;
+use virt_autotest::utils qw(is_xen_host start_guests check_port_state check_host_health check_guest_health);
+use virt_utils qw(collect_host_and_guest_logs);
+
+sub prepare_run_test {
+    reconnect_when_ssh_console_broken unless defined(script_run("rm -f /root/{commands_history,commands_failure}", die_on_timeout => 0));
+    script_run("history -c");
+}
+
+sub run_test {
+
+    my $self = shift;
+    my %health_status = ();
+    $health_status{host} = check_host_health;
+    foreach my $guest (keys %virt_autotest::common::guests) {
+        script_run("virsh start $guest", die_on_timeout => 0);
+        if (script_retry("nmap $guest -PN -p ssh | grep open", delay => 2, retry => 60, die => 0) == 0) {
+            $health_status{$guest} = check_guest_health($guest);
+            if ($health_status{$guest} eq 'fail' || get_var('FORCE_UPLOAD_LOGS')) {
+                script_run("ssh root\@$guest 'dmesg > dmesg.txt 2>&1'", die_on_timeout => 0);
+            }
+        }
+        else {
+            $health_status{$guest} = "Unreachable";
+        }
+        record_info("Failed guest", "$guest: $health_status{$guest}", result => 'fail') unless $health_status{$guest} eq 'pass';
+    }
+    my @unhealthy_list = grep { $health_status{$_} ne 'pass' } keys %health_status;
+
+    # Upload logs on x86_64
+    if (is_x86_64 && (get_var('FORCE_UPLOAD_LOGS') || @unhealthy_list != 0)) {
+        script_run("tar cvzf libvirtd.tar.gz /var/log/libvirt", die_on_timeout => 0);
+        if (is_xen_host) {
+            script_run("tar cvzf xen_logs.tar.gz /var/log/xen", die_on_timeout => 0);
+            script_run("xl dmesg > xl_dmesg.txt", die_on_timeout => 0);
+        }
+        script_run("dmesg > dmesg_host.txt", die_on_timeout => 0);
+        my $host_extra_logs = "/root/libvirtd.tar.gz /root/dmesg_host.txt /root/xen_logs.tar.gz xl_dmesg.txt";
+        my $guest_extra_logs = "/root/dmesg.txt";
+        if (get_var('FORCE_UPLOAD_LOGS')) {
+            collect_host_and_guest_logs(join(' ', grep { $_ ne 'host' } keys %health_status), $host_extra_logs, $guest_extra_logs);
+        }
+        else {
+            collect_host_and_guest_logs(join(' ', grep { $_ ne 'host' } @unhealthy_list), $host_extra_logs, $guest_extra_logs);
```

Comment:
I guess you are aware that the uploaded host logs, which are well selected by you, are available in supportconfig log. And you do this because it does not upload unneeded logs, do you? If yes, I would still recommend to upload supportconfig log, because dev well knows that log structure ,and additional info are given in case needed sometimes, and we can have less code blocks for collecting virt logs. Is it okay?

Reply:
supportconfig logs for both host and guest are supposed to be uploaded by collect_host_and_guest_logs. I just upload more virt logs which may be required, such as libvirtd, dmesg and xen hypervisor logs.

What do you mean here? do you mean other virt logs are not necessary? or .... Sorry, I did not get your point well.

Reply:
I mean  libvirtd, dmesg and xen hypervisor logs are in supportconfig log already :)

Reply:
I guess the libvirtd/dmesg and xen hypervisor logs are not complete? they were always be requested in bugs. If you are sure that supportconfig files are enough, I can stop collecting those virt logs from both host and guest.

Reply:
I just checked that libvirtd.log and qemu/ logs are included in supportconfig files, and they are complete. I'll stop colleting them.